flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/parser/JaninoCompiler.java

```diff
@@ -359,6 +407,15 @@ private static Java.Rvalue generateTypeConvertMethod(
             case "VARCHAR":
             case "STRING":
                 return new Java.MethodInvocation(Location.NOWHERE, null, "castToString", atoms);
+            case "TIMESTAMP":
+                List<Java.Rvalue> timestampFunctionParam = new ArrayList<>(Arrays.asList(atoms));
+                timestampFunctionParam.add(
+                        new Java.AmbiguousName(Location.NOWHERE, new String[] {DEFAULT_TIME_ZONE}));
+                return new Java.MethodInvocation(
+                        Location.NOWHERE,
+                        null,
+                        "castToTimestamp",
+                        timestampFunctionParam.toArray(new Java.Rvalue[0]));
```

Comment:
I remember that there's a `ArrayUtils#add(T[], T)` method in Apache commons that doesn't require initializing mutable lists and keeps code clearer. Is it possible to use it here?

Reply:
I tried to use it but I'm hesitated as many lib used it and we may need to deal dependency shade issue.